radiation/python/radsw/taumol_gt4py.py

```diff
@@ -206,6 +258,293 @@ def loadlookupdata(name):
+    layind: FIELD_INT,
+    nspa: Field[gtscript.IJ, type_nbandssw_int],
+    nspb: Field[gtscript.IJ, type_nbandssw_int],
+    ngs: Field[gtscript.IJ, type_nbandssw_int],
+    id0: Field[type_nbandssw_int],
+    id1: Field[type_nbandssw_int],
+    js: FIELD_INT,
+    jsa: FIELD_INT,
+    colm1: FIELD_FLT,
+    colm2: FIELD_FLT,
+    sfluxref01: Field[(DTYPE_FLT, (16, 1, 7))],
+    sfluxref02: Field[(DTYPE_FLT, (16, 5, 2))],
+    sfluxref03: Field[(DTYPE_FLT, (16, 9, 5))],
+    layreffr: Field[type_nbandssw_int],
+    ix1: Field[type_nbandssw_int],
+    ix2: Field[type_nbandssw_int],
+    ibx: Field[type_nbandssw_int],
+    strrat: Field[type_nbandssw_flt],
+    specwt: Field[type_nbandssw_flt],
+):
+    from __externals__ import (
+        nbandssw,
+        NG16,
+        NG17,
+        NG18,
+        NG19,
+        NG20,
+        NG21,
+        NG22,
+        NG23,
+        NG24,
+        NG25,
+        NG26,
+        NG27,
+        NG28,
+        NG29,
+        scalekur,
+        oneminus,
+    )
+
+    with computation(FORWARD), interval(...):
+        if idxday:
+            for jb in range(nbandssw):
+                if laytrop:
+                    id0[0, 0, 0][jb] = ((jp - 1) * 5 + (jt - 1)) * nspa[0, 0][jb]
+                    id1[0, 0, 0][jb] = (jp * 5 + (jt1 - 1)) * nspa[0, 0][jb]
+                else:
+                    id0[0, 0, 0][jb] = ((jp - 13) * 5 + (jt - 1)) * nspb[0, 0][jb]
+                    id1[0, 0, 0][jb] = ((jp - 12) * 5 + (jt1 - 1)) * nspb[0, 0][jb]
+
+            for j in range(NG16):
+                sfluxzen[0, 0][ngs[0, 0][0] + j] = sfluxref01[0, 0, 0][
+                    j, 0, ibx[0, 0, 0][0]
+                ]
+            for j2 in range(NG20):
+                sfluxzen[0, 0][ngs[0, 0][4] + j2] = sfluxref01[0, 0, 0][
+                    j2, 0, ibx[0, 0, 0][4]
+                ]
+            for j3 in range(NG23):
+                sfluxzen[0, 0][ngs[0, 0][7] + j3] = sfluxref01[0, 0, 0][
+                    j3, 0, ibx[0, 0, 0][7]
+                ]
+            for j4 in range(NG25):
+                sfluxzen[0, 0][ngs[0, 0][9] + j4] = sfluxref01[0, 0, 0][
+                    j4, 0, ibx[0, 0, 0][9]
+                ]
+            for j5 in range(NG26):
+                sfluxzen[0, 0][ngs[0, 0][10] + j5] = sfluxref01[0, 0, 0][
+                    j5, 0, ibx[0, 0, 0][10]
+                ]
+            for j6 in range(NG29):
+                sfluxzen[0, 0][ngs[0, 0][13] + j6] = sfluxref01[0, 0, 0][
+                    j6, 0, ibx[0, 0, 0][13]
+                ]
+
+            for j7 in range(NG27):
+                sfluxzen[0, 0][ngs[0, 0][11] + j7] = (
+                    scalekur * sfluxref01[0, 0, 0][j7, 0, ibx[0, 0, 0][11]]
+                )
+
+    with computation(FORWARD), interval(0, -1):
+        if idxday:
+            if not laytrop:
+                # case default
+                cond = jp < layreffr[0, 0, 0][1] and jp[0, 0, 1] >= layreffr[0, 0, 0][1]
```

Comment:
Yes, that's fine because all these arrays are padded with an extra 0, so all the actual data is still included. The output array is 2D so it doesn't matter that the previous computation is over one more index